src/main/java/prefs/Preferences.java

```diff
@@ -10,160 +16,296 @@
+    private final SessionConfig sessionConfig;
+    private final UserConfig userConfig;
+
+    /**
+     * Initialises a Preferences instance that contains the configurations in the config files specified.
+     * @param configDirectory The directory that the config files are held in.
+     * @param sessionConfigFileName The name of the session config file to load.
+     * @param userConfigFileName The name of the user config file to load.
+     * @return The initialised Preferences instance.
+     */
+    public static Preferences load(String configDirectory, String sessionConfigFileName, String userConfigFileName) {
+        return new Preferences(configDirectory, sessionConfigFileName, userConfigFileName, false);
+    }
 
-    private Preferences(String configFileName, boolean createUnconditionally) {
-        this.fileHandler = new ConfigFileHandler(DIRECTORY, configFileName);
+    /**
+     * Initialises a Preferences instance that creates new config files.
+     * @param configDirectory The directory that the config files are held in.
+     * @param sessionConfigFileName The name of the session config file to create.
+     * @param userConfigFileName The name of the user config file to create.
+     * @return The initialised Preferences instance.
+     */
+    public static Preferences create(String configDirectory, String sessionConfigFileName, String userConfigFileName) {
+        return new Preferences(configDirectory, sessionConfigFileName, userConfigFileName, true);
+    }
 
-        if (createUnconditionally) {
-            initGlobalConfig();
+    /**
+     * @param configDirectory The directory that the config files are held in.
+     * @param sessionConfigFileName The name of the session config file
+     * @param userConfigFileName The name of the user config file
+     * @param ignoreExisting True if existing config file is to be ignored.
+     */
+    private Preferences(String configDirectory,
+                        String sessionConfigFileName,
+                        String userConfigFileName,
+                        boolean ignoreExisting) {
+        this.configDirectory = configDirectory;
+        this.sessionConfigFileName = sessionConfigFileName;
+        this.userConfigFileName = userConfigFileName;
+
+        if (ignoreExisting) {
+            this.sessionConfig = createConfig(DEFAULT_FILE_CONTENTS, SessionConfig.class);
+            this.userConfig = createConfig(DEFAULT_FILE_CONTENTS, UserConfig.class);
         } else {
-            loadGlobalConfig();
+            this.sessionConfig = loadConfig(configDirectory, sessionConfigFileName, SessionConfig.class);
+            this.userConfig = loadConfig(configDirectory, userConfigFileName, UserConfig.class);
         }
     }
 
     /**
-     * Initialises a Preferences instance which creates its config file, or loads
-     * from it if it already exists.
+     * Creates a new Config instance
+     * @param contents The contents of the object to create
```

Comment:
Would be good to say that this should be JSON string or JSON representation of an object since contents of object is very ambiguous.